### DISCUSSION TITLE
Support for multiple data sources

### DISCUSSION THREAD

**Discussion by nyemiks** [openmdx/openmdx#32] [Q&A]
I had a preliminary discussion with a prospect some time ago and they told us they will not want to share the same database with other tenants.

My current understanding is that openmdx partitions data at two levels:

1. The provider level.

2. The Segment level.

To accomodate this prospect requirements. I will need to provide support for at least one of the following multi tenant
database options:

1. One Tenant per Schema. In this scenario, a database schema  is created for the tenant with the application's tables. This tenant will have exclusive access to it's tables.

2. One Tenant per database. In this scenario, a database is created for the tenant with the application's tables. This tenant will have exclusive access to the database.

In some sectors, e.g. finance sector option 2 might be the only acceptable option.

At the application server level. The above is resolved by creating data sources which encapsulates the database connection to each schema or database.

I will like to ask how openmdx supports the above scenario's ?

Can I register multiple data sources in openmdx ? If yes how can I configure it in openmdx ?

If this is not possible what are the work arounds ?

**Comment by DlRTY-HARRY** [comment]
Yes, multiple DataSource connections are supported by openMDX:

1. define multiple dataproviders in the jdoconfig.xml file
2. configure a specific DataSource for each dataprovider, e.g. `PERSISTENCE/datasourceName=java:comp/env/jdbc/DataSource1` and `PERSISTENCE/datasourceName=java:comp/env/jdbc/DataSource2`, respectively
